include/libtorrent/config.hpp

```diff
@@ -134,13 +134,6 @@ POSSIBILITY OF SUCH DAMAGE.
 # endif
 #include <AvailabilityMacros.h>
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
-// on OSX, use the built-in common crypto for built-in
-# if !defined TORRENT_USE_OPENSSL && !defined TORRENT_USE_GCRYPT
-#  define TORRENT_USE_COMMONCRYPTO 1
```

Comment:
I'm sorry. but I think it's nicer to detect this automatically here. This seems to provide the semantics I proposed above. If we're using built-in crypto on a mac, use the official API.